src/python/gem5/simulate/simulator.py

```diff
@@ -574,8 +690,23 @@ def run(self, max_ticks: int = m5.MaxTick) -> None:
                           exit event is received, if another simulation exit
                           event is met the tick count is reset. This is the
                           **maximum number of ticks per simulation run**.
+                          **WARNING**: This this parameter is deprecated and
+                          will be removed in future releases of gem5. Please
+                          set the max ticks via the `set_max_ticks` function.
+                          or via the Simulator constructor.
```

Comment:
I disagree with deprecating this API. 

I suggest keeping this API and doing a `warn` if the `_max_ticks` is *changed* when calling `run` from when the simulator was originally instantiated or if set via `set_max_ticks`. 

I think the *most common* use case will be to set the max ticks in the `run` function.